commands/alpha/rpkg/util/common.go

```diff
@@ -67,43 +68,33 @@ func GetResourceFileKubeObject(prr *api.PackageRevisionResources, file, kind, na
 	return ko, nil
 }
 
-func GetResourceVersionAnnotation(prr *api.PackageRevisionResources) (string, error) {
-	ko, err := GetResourceFileKubeObject(prr, "Kptfile", "Kptfile", "")
-
+func GetResourceVersion(prr *api.PackageRevisionResources) (string, error) {
+	ko, err := GetResourceFileKubeObject(prr, kptfilev1.RevisionMetaDataFileName, kptfilev1.RevisionMetaDataKind, "")
 	if err != nil {
 		return "", err
 	}
-	annotations := ko.GetAnnotations()
-	rv, ok := annotations[ResourceVersionAnnotation]
-	if !ok {
-		rv = ""
-	}
+	rv, _, _ := ko.NestedString("metadata", "resourceVersion")
 	return rv, nil
 }
 
-func AddResourceVersionAnnotation(prr *api.PackageRevisionResources) error {
-	ko, err := GetResourceFileKubeObject(prr, "Kptfile", "Kptfile", "")
+func AddRevisionMetaData(prr *api.PackageRevisionResources) error {
+	kptfileKo, err := GetResourceFileKubeObject(prr, "Kptfile", "Kptfile", "")
 	if err != nil {
 		return err
 	}
 
-	ko.SetAnnotation(ResourceVersionAnnotation, prr.GetResourceVersion())
-	prr.Spec.Resources["Kptfile"] = ko.String()
+	kptMetaDataKo := fnsdk.NewEmptyKubeObject()
+	kptMetaDataKo.SetAPIVersion(kptfileKo.GetAPIVersion())
```

Comment:
Rather then setting this to the API version of the Kptfile, we probably should set it to the API version of the PackageRevisionResource. Then you don't even need the `kptfileKo`, and it more accurately reflects that this is related to the PRR not the Kptfile.

Reply:
Agreed, Just wanted to align the contents with the Kptfile intially. But it makes sense to remove it.